client/src/components/podcasts/PodcastTile.js

```diff
@@ -1,20 +1,167 @@
+  }
+
+  const newVote = async (votePayload) => {
+    try {
+      const response = await fetch(`/api/v1/podcasts/${podcast.id}/votes`, {
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify(votePayload)
+      })
+      if (!response.ok) {
+        throw new Error(`${response.status} (${response.statusText})`)
+      }
+      const body = await response.json()
+      const votes = body.votes
+      const voteType = votes.userVotes.value === 1 ? 'up' : 'down'
+      if (voteType === 'up') {
+        setUserVote({ hasUpVoted: true, hasDownVoted: false })
+      } else {
+        setUserVote({ hasUpVoted: false, hasDownVoted: true })
+      }
+      setTotalVotes(votes.totalVotes.value)
+    } catch (error) {
+      console.error(error.message)
+    }
+  }
+
+  const editVote = async (votePayload) => {
+    try {
+      const response = await fetch(`/api/v1/podcasts/${podcast.id}/votes`, {
+        method: "PATCH",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify(votePayload)
+      })
+      if (!response.ok) {
+        throw new Error(`${response.status} (${response.statusText})`)
+      }
+      const body = await response.json()
+      const votes = body.votes
+      if (votes.userVotes.value === 1) {
+        setUserVote({ hasUpVoted: true, hasDownVoted: false })
+      } else {
+        setUserVote({ hasUpVoted: false, hasDownVoted: true })
+      }
+      setTotalVotes(votes.totalVotes.value)
+    } catch (error) {
+      console.error(error.message)
+    }
+  }
+
+  const removeVote = async (votePayload) => {
+    try {
+      const response = await fetch(`/api/v1/podcasts/${podcast.id}/votes`, {
+        method: "DELETE",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify(votePayload)
+      })
+      if (!response.ok) {
+        throw new Error(`${response.status} (${response.statusText})`)
+      }
+      const body = await response.json()
+      const votes = body.votes
+      setTotalVotes(votes.totalVotes.value)
+      setUserVote({ hasUpVoted: false, hasDownVoted: false })
+    } catch (error) {
+      console.error(error.message)
+    }
+  }
+
+  const upVoteClickHandler = (event) => {
+    event.preventDefault()
+    if (userVotes.hasDownVoted && !userVotes.hasUpVoted) {
+      editVote({ value: 1 })
+    } else if (userVotes.hasUpVoted) {
+      removeVote({ value: 0 })
+    } else {
+      newVote({ value: 1 })
+    }
+  }
 
-  if (user !== null && user.id === podcast.userId) {
-    editDeleteButtons = (
+  const downVoteClickHandler = (event) => {
+    event.preventDefault()
+    if (!userVotes.hasDownVoted && userVotes.hasUpVoted) {
+      editVote({ value: -1 })
+    } else if (userVotes.hasDownVoted) {
+      removeVote({ value: 0 })
+    } else {
+      newVote({ value: -1 })
+    }
+  }
```

Comment:
ill not require this, but these vote handlers have so much logic, that it would serve to make a seperate function or object that isnt a component to help manage this logic. but that would be an extensive refactor